components/sts/org.wso2.carbon.identity.sts.passive.ui/src/main/java/org/wso2/carbon/identity/sts/passive/ui/PassiveSTS.java

```diff
@@ -195,11 +196,11 @@ private void sendData(HttpServletResponse httpResp, ResponseToken respToken, Str
         } else {
             finalPage = pageWithReplyActionResultContext.replace("<!--$additionalParams-->",
                     "<input type='hidden' name='AuthenticatedIdPs' value='" +
-                            URLEncoder.encode(authenticatedIdPs, "UTF-8") + "'>");
+                            Encode.forHtmlAttribute(authenticatedIdPs) + "'>");
         }
 
         PrintWriter out = httpResp.getWriter();
-        out.print(finalPage);
+        out.print(Encode.forHtml(finalPage));
```

Comment:
It's wrong to encode the full page here. Instead encode replacement for $url (line 182). Anyhow I couldn't see $action or $result in the html page template at IS_HOME/repository/resources/identity/pages/sts_response.html. Check if those are needed and remove if no longer used. Further encode line 189